payload/vanguard/handler/upload.py

```diff
@@ -0,0 +1,56 @@
+from ..protocol.vanguard import ProgramUploadMsg, ProgramResultMsg
+
+import vanguard.protocol.vanguard
+from ..radio import Radio
+import struct
+import subprocess
+
+
+
+class UploadHandler(object):
+    msg_type = ProgramUploadMsg.TYPE
+    
+    def handle(self, radio, msg):
+        self.radio = radio
+        self.parse_message(msg)
+
+    def parse_message(self, msg):
+        fmt = '!HLBBLHHHHH' #index, chunk, chunk_count, program_name_len, program_data_len
+        unpacked_data = struct.unpack_from(fmt, msg.as_buffer()) 
+        self.index = unpacked_data[5]
```

Comment:
hrm, what are the first 4 entries in the struct (`HLBBL`?) are those the vanguard message parts?